File: lib/src/screens/personal_info_screen.dart
import 'package:flutter/material.dart';
import '../constants/app_color.dart';
import '../models/user.dart';
import '../widgets/components/gradient_button.dart';

class PersonalInfoScreen extends StatefulWidget {
  final User? user;

  const PersonalInfoScreen({
    Key? key,
    this.user,
  }) : super(key: key);

  @override
  State<PersonalInfoScreen> createState() => _PersonalInfoScreenState();
}

class _PersonalInfoScreenState extends State<PersonalInfoScreen>
    with SingleTickerProviderStateMixin {
  final _scrollController = ScrollController();
  bool _showContent = false;
  bool _showSuccessMessage = false;

  // Form controllers
  late TextEditingController _firstNameController;
  late TextEditingController _lastNameController;
  late TextEditingController _emailController;
  late TextEditingController _phoneController;
  late TextEditingController _addressController;
  late TextEditingController _cityController;
  late TextEditingController _stateController;
  late TextEditingController _zipCodeController;

  @override
  void initState() {
    super.initState();

    // Initialize controllers with user data
    _firstNameController =
        TextEditingController(text: widget.user?.firstName ?? "");
    _lastNameController =
        TextEditingController(text: widget.user?.lastName ?? "");
    _emailController = TextEditingController(text: widget.user?.email ?? "");
    _phoneController = TextEditingController(text: widget.user?.phone ?? "");
    _addressController =
        TextEditingController(text: widget.user?.address ?? "");
    _cityController = TextEditingController(text: widget.user?.city ?? "");
    _stateController = TextEditingController(text: widget.user?.state ?? "");
    _zipCodeController =
        TextEditingController(text: widget.user?.zipCode ?? "");

    // Trigger animation after a delay
    Future.delayed(const Duration(milliseconds: 100), () {
      if (mounted) {
        setState(() {
          _showContent = true;
        });
      }
    });
  }

  @override
  void dispose() {
    _firstNameController.dispose();
    _lastNameController.dispose();
    _emailController.dispose();
    _phoneController.dispose();
    _addressController.dispose();
    _cityController.dispose();
    _stateController.dispose();
    _zipCodeController.dispose();
    _scrollController.dispose();
    super.dispose();
  }

  // Check if form has changes
  bool get _hasChanges {
    return _firstNameController.text != (widget.user?.firstName ?? "") ||
        _lastNameController.text != (widget.user?.lastName ?? "") ||
        _emailController.text != (widget.user?.email ?? "") ||
        _phoneController.text != (widget.user?.phone ?? "") ||
        _addressController.text != (widget.user?.address ?? "") ||
        _cityController.text != (widget.user?.city ?? "") ||
        _stateController.text != (widget.user?.state ?? "") ||
        _zipCodeController.text != (widget.user?.zipCode ?? "");
  }

  // Check if form is valid
  bool get _isFormValid {
    return _firstNameController.text.isNotEmpty &&
        _lastNameController.text.isNotEmpty &&
        _emailController.text.isNotEmpty &&
        _emailController.text.contains('@') &&
        _phoneController.text.isNotEmpty;
  }

  void _saveChanges() {
    // Mock saving changes
    setState(() {
      _showSuccessMessage = true;
    });

    // Hide success message after a delay
    Future.delayed(const Duration(seconds: 3), () {
      if (mounted) {
        setState(() {
          _showSuccessMessage = false;
        });
      }
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text(
          'Personal Information',
          style: TextStyle(
            fontSize: 24,
            fontWeight: FontWeight.w600,
            color: AppColor.textDarkBlue,
          ),
        ),
        backgroundColor: Colors.transparent,
        elevation: 0,
        iconTheme: const IconThemeData(color: AppColor.textDarkBlue),
      ),
      body: Container(
        width: double.infinity,
        height: double.infinity,
        decoration: const BoxDecoration(
          gradient: LinearGradient(
            begin: Alignment.topCenter,
            end: Alignment.bottomCenter,
            colors: [
              Color(0xFFF8FAFC),
              Color(0xFFF5F7FD),
            ],
          ),
        ),
        child: Stack(
          children: [
            // Content
            AnimatedOpacity(
              opacity: _showContent ? 1.0 : 0.0,
              duration: const Duration(milliseconds: 500),
              curve: Curves.easeOut,
              child: AnimatedSlide(
                offset: _showContent ? Offset.zero : const Offset(0, 0.05),
                duration: const Duration(milliseconds: 500),
                curve: Curves.easeOutCubic,
                child: SingleChildScrollView(
                  controller: _scrollController,
                  padding:
                      const EdgeInsets.symmetric(horizontal: 20, vertical: 16),
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.center,
                    children: [
                      // Personal Information Card
                      Card(
                        shape: RoundedRectangleBorder(
                          borderRadius: BorderRadius.circular(24),
                        ),
                        elevation: 4,
                        child: Padding(
                          padding: const EdgeInsets.all(24),
                          child: Column(
                            crossAxisAlignment: CrossAxisAlignment.start,
                            children: [
                              const Text(
                                'Edit Your Information',
                                style: TextStyle(
                                  fontSize: 20,
                                  fontWeight: FontWeight.bold,
                                  color: AppColor.textDarkBlue,
                                ),
                              ),

                              const SizedBox(height: 16),

                              // Name fields (side by side)
                              Row(
                                children: [
                                  // First Name
                                  Expanded(
                                    child: _buildFormField(
                                      controller: _firstNameController,
                                      label: 'First Name',
                                      icon: Icons.person_outline,
                                      textCapitalization:
                                          TextCapitalization.words,
                                    ),
                                  ),

                                  const SizedBox(width: 16),

                                  // Last Name
                                  Expanded(
                                    child: _buildFormField(
                                      controller: _lastNameController,
                                      label: 'Last Name',
                                      icon: Icons.person_outline,
                                      textCapitalization:
                                          TextCapitalization.words,
                                    ),
                                  ),
                                ],
                              ),

                              const SizedBox(height: 16),

                              // Email
                              _buildFormField(
                                controller: _emailController,
                                label: 'Email',
                                icon: Icons.email_outlined,
                                keyboardType: TextInputType.emailAddress,
                              ),

                              const SizedBox(height: 16),

                              // Phone
                              _buildFormField(
                                controller: _phoneController,
                                label: 'Phone',
                                icon: Icons.phone_outlined,
                                keyboardType: TextInputType.phone,
                              ),

                              const SizedBox(height: 16),

                              // Address
                              _buildFormField(
                                controller: _addressController,
                                label: 'Street Address',
                                icon: Icons.home_outlined,
                                textCapitalization: TextCapitalization.words,
                              ),

                              const SizedBox(height: 16),

                              // City, State, Zip (side by side)
                              Row(
                                crossAxisAlignment: CrossAxisAlignment.start,
                                children: [
                                  // City
                                  Expanded(
                                    flex: 2,
                                    child: _buildFormField(
                                      controller: _cityController,
                                      label: 'City',
                                      icon: Icons.location_city_outlined,
                                      textCapitalization:
                                          TextCapitalization.words,
                                    ),
                                  ),

                                  const SizedBox(width: 16),

                                  // State
                                  Expanded(
                                    child: _buildFormField(
                                      controller: _stateController,
                                      label: 'State',
                                      icon: Icons.map_outlined,
                                      textCapitalization:
                                          TextCapitalization.characters,
                                    ),
                                  ),

                                  const SizedBox(width: 16),

                                  // Zip Code
                                  Expanded(
                                    child: _buildFormField(
                                      controller: _zipCodeController,
                                      label: 'Zip',
                                      icon: Icons.pin_outlined,
                                      keyboardType: TextInputType.number,
                                    ),
                                  ),
                                ],
                              ),
                            ],
                          ),
                        ),
                      ),

                      const SizedBox(height: 24),

                      // Save Button
                      GradientButton(
                        text: 'Save Changes',
                        onPressed: _isFormValid && _hasChanges
                            ? () => _saveChanges()
                            : () {},
                        enabled: _isFormValid && _hasChanges,
                      ),
                    ],
                  ),
                ),
              ),
            ),

            // Success message
            if (_showSuccessMessage)
              Positioned(
                top: 16,
                left: 16,
                right: 16,
                child: Container(
                  padding:
                      const EdgeInsets.symmetric(horizontal: 16, vertical: 12),
                  decoration: BoxDecoration(
                    color: AppColor.successGreen.withOpacity(0.9),
                    borderRadius: BorderRadius.circular(8),
                    boxShadow: [
                      BoxShadow(
                        color: Colors.black.withOpacity(0.1),
                        blurRadius: 8,
                        offset: const Offset(0, 2),
                      ),
                    ],
                  ),
                  child: const Row(
                    children: [
                      Icon(
                        Icons.check_circle,
                        color: Colors.white,
                      ),
                      SizedBox(width: 12),
                      Expanded(
                        child: Text(
                          'Your information has been updated successfully!',
                          style: TextStyle(
                            color: Colors.white,
                            fontWeight: FontWeight.w500,
                          ),
                        ),
                      ),
                    ],
                  ),
                ),
              ),
          ],
        ),
      ),
    );
  }

  Widget _buildFormField({
    required TextEditingController controller,
    required String label,
    required IconData icon,
    TextInputType keyboardType = TextInputType.text,
    TextCapitalization textCapitalization = TextCapitalization.none,
  }) {
    return TextField(
      controller: controller,
      keyboardType: keyboardType,
      textCapitalization: textCapitalization,
      decoration: InputDecoration(
        labelText: label,
        prefixIcon: Icon(icon, color: AppColor.primaryBlue),
        border: OutlineInputBorder(
          borderRadius: BorderRadius.circular(12),
          borderSide: const BorderSide(color: Color(0xFFE5E7EB)),
        ),
        enabledBorder: OutlineInputBorder(
          borderRadius: BorderRadius.circular(12),
          borderSide: const BorderSide(color: Color(0xFFE5E7EB)),
        ),
        focusedBorder: OutlineInputBorder(
          borderRadius: BorderRadius.circular(12),
          borderSide: const BorderSide(color: AppColor.primaryBlue, width: 2),
        ),
        filled: true,
        fillColor: Colors.white,
        contentPadding:
            const EdgeInsets.symmetric(vertical: 16, horizontal: 16),
      ),
      style: const TextStyle(
        fontSize: 16,
        color: AppColor.textDarkBlue,
      ),
    );
  }
}

File: backup/kotlin/ui/PersonalInfoScreen.kt
package com.example.healthride.ui

import androidx.compose.animation.*
import androidx.compose.animation.core.*
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material.icons.outlined.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.ImeAction
import androidx.compose.ui.text.input.KeyboardCapitalization
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.unit.dp
import com.example.healthride.data.model.User
import com.example.healthride.ui.theme.*
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun PersonalInfoScreen(
    onBackClick: () -> Unit = {},
    // Sample user data for preview
    user: User? = User(
        id = "user_123",
        email = "user@example.com",
        firstName = "John",
        lastName = "Doe",
        phone = "(555) 123-4567",
        address = "123 Health Street",
        city = "Medical City",
        state = "MC",
        zipCode = "12345"
    )
) {
    val scrollState = rememberScrollState()
    val coroutineScope = rememberCoroutineScope()
    var showSuccessMessage by remember { mutableStateOf(false) }

    // Animation states
    var showContent by remember { mutableStateOf(false) }

    // Form states (initialized with user data if available)
    var firstName by remember { mutableStateOf(user?.firstName ?: "") }
    var lastName by remember { mutableStateOf(user?.lastName ?: "") }
    var email by remember { mutableStateOf(user?.email ?: "") }
    var phone by remember { mutableStateOf(user?.phone ?: "") }
    var address by remember { mutableStateOf(user?.address ?: "") }
    var city by remember { mutableStateOf(user?.city ?: "") }
    var state by remember { mutableStateOf(user?.state ?: "") }
    var zipCode by remember { mutableStateOf(user?.zipCode ?: "") }

    // Track if form has changes
    val hasChanges = remember(firstName, lastName, email, phone, address, city, state, zipCode) {
        firstName != user?.firstName || lastName != user?.lastName ||
                email != user?.email || phone != user?.phone ||
                address != user?.address || city != user?.city ||
                state != user?.state || zipCode != user?.zipCode
    }

    // Form validation
    val isFormValid = remember(firstName, lastName, email, phone) {
        firstName.isNotBlank() && lastName.isNotBlank() &&
                email.isNotBlank() && email.contains("@") && phone.isNotBlank()
    }

    // Start animations
    LaunchedEffect(Unit) {
        delay(100)
        showContent = true
    }

    // Hide success message after delay
    LaunchedEffect(showSuccessMessage) {
        if (showSuccessMessage) {
            delay(3000)
            showSuccessMessage = false
        }
    }

    Scaffold(
        topBar = {
            TopAppBar(
                title = {
                    Text(
                        "Personal Information",
                        fontWeight = FontWeight.SemiBold
                    )
                },
                navigationIcon = {
                    IconButton(onClick = onBackClick) {
                        Icon(
                            imageVector = Icons.Default.ArrowBack,
                            contentDescription = "Back",
                            tint = TextDarkBlue
                        )
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = Color.Transparent,
                    titleContentColor = TextDarkBlue
                )
            )
        }
    ) { paddingValues ->
        Box(
            modifier = Modifier
                .fillMaxSize()
                .background(
                    brush = Brush.verticalGradient(
                        colors = listOf(
                            Color(0xFFF8FAFC),
                            Color(0xFFF5F7FD)
                        )
                    )
                )
        ) {
            AnimatedVisibility(
                visible = showContent,
                enter = fadeIn() + slideInVertically(initialOffsetY = { it / 4 }),
                modifier = Modifier
                    .fillMaxSize()
                    .padding(paddingValues)
            ) {
                Column(
                    modifier = Modifier
                        .fillMaxSize()
                        .verticalScroll(scrollState)
                        .padding(horizontal = 20.dp, vertical = 16.dp),
                    horizontalAlignment = Alignment.CenterHorizontally,
                    verticalArrangement = Arrangement.Center
                ) {
                    // Personal Information Card
                    Card(
                        modifier = Modifier.fillMaxWidth(),
                        shape = RoundedCornerShape(24.dp),
                        colors = CardDefaults.cardColors(
                            containerColor = SurfaceWhite
                        ),
                        elevation = CardDefaults.cardElevation(
                            defaultElevation = 4.dp
                        )
                    ) {
                        Column(
                            modifier = Modifier.padding(24.dp),
                            verticalArrangement = Arrangement.spacedBy(16.dp)
                        ) {
                            Text(
                                text = "Edit Your Information",
                                style = MaterialTheme.typography.titleLarge,
                                fontWeight = FontWeight.Bold,
                                color = TextDarkBlue
                            )

                            Spacer(modifier = Modifier.height(8.dp))

                            // Name fields (side by side)
                            Row(
                                modifier = Modifier.fillMaxWidth(),
                                horizontalArrangement = Arrangement.spacedBy(16.dp)
                            ) {
                                // First Name
                                Column(modifier = Modifier.weight(1f)) {
                                    FormField(
                                        value = firstName,
                                        onValueChange = { firstName = it },
                                        label = "First Name",
                                        icon = Icons.Outlined.Person,
                                        keyboardOptions = KeyboardOptions(
                                            capitalization = KeyboardCapitalization.Words,
                                            imeAction = ImeAction.Next
                                        )
                                    )
                                }

                                // Last Name
                                Column(modifier = Modifier.weight(1f)) {
                                    FormField(
                                        value = lastName,
                                        onValueChange = { lastName = it },
                                        label = "Last Name",
                                        icon = Icons.Outlined.Person,
                                        keyboardOptions = KeyboardOptions(
                                            capitalization = KeyboardCapitalization.Words,
                                            imeAction = ImeAction.Next
                                        )
                                    )
                                }
                            }

                            // Contact Information
                            FormSection(title = "Contact Information") {
                                FormField(
                                    value = email,
                                    onValueChange = { email = it },
                                    label = "Email Address",
                                    icon = Icons.Outlined.Email,
                                    keyboardOptions = KeyboardOptions(
                                        keyboardType = KeyboardType.Email,
                                        imeAction = ImeAction.Next
                                    )
                                )

                                FormField(
                                    value = phone,
                                    onValueChange = { phone = it },
                                    label = "Phone Number",
                                    icon = Icons.Outlined.Phone,
                                    keyboardOptions = KeyboardOptions(
                                        keyboardType = KeyboardType.Phone,
                                        imeAction = ImeAction.Next
                                    )
                                )
                            }

                            // Address Information
                            FormSection(title = "Address Information") {
                                FormField(
                                    value = address,
                                    onValueChange = { address = it },
                                    label = "Street Address",
                                    icon = Icons.Outlined.Home,
                                    keyboardOptions = KeyboardOptions(
                                        capitalization = KeyboardCapitalization.Words,
                                        imeAction = ImeAction.Next
                                    )
                                )

                                FormField(
                                    value = city,
                                    onValueChange = { city = it },
                                    label = "City",
                                    icon = Icons.Outlined.LocationCity,
                                    keyboardOptions = KeyboardOptions(
                                        capitalization = KeyboardCapitalization.Words,
                                        imeAction = ImeAction.Next
                                    )
                                )

                                // State and Zip code (side by side)
                                Row(
                                    modifier = Modifier.fillMaxWidth(),
                                    horizontalArrangement = Arrangement.spacedBy(16.dp)
                                ) {
                                    // State
                                    Column(modifier = Modifier.weight(1f)) {
                                        FormField(
                                            value = state,
                                            onValueChange = { state = it },
                                            label = "State",
                                            icon = Icons.Outlined.Map,
                                            keyboardOptions = KeyboardOptions(
                                                capitalization = KeyboardCapitalization.Characters,
                                                imeAction = ImeAction.Next
                                            )
                                        )
                                    }

                                    // Zip Code
                                    Column(modifier = Modifier.weight(1f)) {
                                        FormField(
                                            value = zipCode,
                                            onValueChange = { zipCode = it },
                                            label = "Zip Code",
                                            icon = Icons.Outlined.Pin,
                                            keyboardOptions = KeyboardOptions(
                                                keyboardType = KeyboardType.Number,
                                                imeAction = ImeAction.Done
                                            )
                                        )
                                    }
                                }
                            }
                        }
                    }

                    Spacer(modifier = Modifier.height(24.dp))

                    // Save Button
                    Button(
                        onClick = {
                            // Simulate saving
                            coroutineScope.launch {
                                showSuccessMessage = true
                                // In a real app, you would save to database here
                            }
                        },
                        modifier = Modifier
                            .fillMaxWidth()
                            .height(56.dp),
                        shape = RoundedCornerShape(28.dp),
                        colors = ButtonDefaults.buttonColors(
                            containerColor = PrimaryBlue
                        ),
                        enabled = hasChanges && isFormValid
                    ) {
                        Icon(
                            imageVector = Icons.Filled.Save,
                            contentDescription = null,
                            modifier = Modifier.size(20.dp)
                        )

                        Spacer(modifier = Modifier.width(8.dp))

                        Text(
                            text = "Save Changes",
                            fontWeight = FontWeight.Bold
                        )
                    }

                    Spacer(modifier = Modifier.height(24.dp))

                    // If no changes, show a helper text
                    if (!hasChanges) {
                        Text(
                            text = "Make changes to your information, then tap Save.",
                            style = MaterialTheme.typography.bodyMedium,
                            color = TextMediumGray
                        )
                    }
                    else if (!isFormValid) {
                        Text(
                            text = "Please fill in all required fields.",
                            style = MaterialTheme.typography.bodyMedium,
                            color = ErrorRed
                        )
                    }
                }
            }

            // Success Message Snackbar
            AnimatedVisibility(
                visible = showSuccessMessage,
                enter = slideInVertically(initialOffsetY = { it }) + fadeIn(),
                exit = slideOutVertically(targetOffsetY = { it }) + fadeOut(),
                modifier = Modifier
                    .padding(16.dp)
                    .align(Alignment.BottomCenter)
            ) {
                Surface(
                    color = SuccessGreen,
                    shape = RoundedCornerShape(8.dp),
                    shadowElevation = 4.dp
                ) {
                    Row(
                        modifier = Modifier
                            .padding(horizontal = 16.dp, vertical = 12.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Icon(
                            imageVector = Icons.Filled.CheckCircle,
                            contentDescription = null,
                            tint = Color.White
                        )

                        Spacer(modifier = Modifier.width(12.dp))

                        Text(
                            text = "Changes saved successfully!",
                            color = Color.White,
                            fontWeight = FontWeight.Medium
                        )
                    }
                }
            }
        }
    }
}

@Composable
fun FormSection(
    title: String,
    content: @Composable () -> Unit
) {
    Column(
        modifier = Modifier.fillMaxWidth(),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        Row(
            verticalAlignment = Alignment.CenterVertically
        ) {
            Divider(
                modifier = Modifier.weight(0.2f),
                color = PrimaryBlue.copy(alpha = 0.3f),
                thickness = 2.dp
            )

            Text(
                text = title,
                style = MaterialTheme.typography.titleMedium,
                fontWeight = FontWeight.Bold,
                color = TextDarkBlue,
                modifier = Modifier.padding(horizontal = 8.dp)
            )

            Divider(
                modifier = Modifier.weight(1f),
                color = PrimaryBlue.copy(alpha = 0.2f),
                thickness = 1.dp
            )
        }

        content()
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun FormField(
    value: String,
    onValueChange: (String) -> Unit,
    label: String,
    icon: ImageVector,
    keyboardOptions: KeyboardOptions = KeyboardOptions.Default,
    isError: Boolean = false,
    errorMessage: String = ""
) {
    Column(
        modifier = Modifier.fillMaxWidth()
    ) {
        OutlinedTextField(
            value = value,
            onValueChange = onValueChange,
            label = { Text(label) },
            leadingIcon = {
                Box(
                    modifier = Modifier
                        .size(36.dp)
                        .clip(CircleShape)
                        .background(PrimaryBlue.copy(alpha = 0.1f)),
                    contentAlignment = Alignment.Center
                ) {
                    Icon(
                        imageVector = icon,
                        contentDescription = null,
                        tint = PrimaryBlue,
                        modifier = Modifier.size(20.dp)
                    )
                }
            },
            modifier = Modifier.fillMaxWidth(),
            shape = RoundedCornerShape(12.dp),
            colors = OutlinedTextFieldDefaults.colors(
                focusedBorderColor = PrimaryBlue,
                unfocusedBorderColor = TextLightGray,
                focusedLabelColor = PrimaryBlue,
                cursorColor = PrimaryBlue,
                focusedContainerColor = Color.White,
                unfocusedContainerColor = Color.White,
                errorBorderColor = ErrorRed,
                errorLabelColor = ErrorRed,
                errorCursorColor = ErrorRed
            ),
            keyboardOptions = keyboardOptions,
            isError = isError,
            singleLine = true
        )

        // Error message
        if (isError && errorMessage.isNotBlank()) {
            Text(
                text = errorMessage,
                style = MaterialTheme.typography.bodySmall,
                color = ErrorRed,
                modifier = Modifier.padding(start = 16.dp, top = 4.dp)
            )
        }
    }
}